semester_1/kr2.1/task2/graph/graph.c

```diff
@@ -0,0 +1,111 @@
+#include <stdbool.h>
+
+#include "graph.h"
+#include "../list/list.h"
+
+typedef struct Edge
+{
+    size_t firstNodeNumber;
+    size_t secondNodeNumber;
+} Edge;
+
+typedef struct Graph
+{
+    size_t edgeAmount;
+    Edge **edges;
+} Graph;
+
+Graph *createGraph(const size_t nodeAmount, const size_t edgeAmount)
+{
+    Graph *graph = calloc(1, sizeof(Graph));
+    if (graph == NULL)
+    {
+        return NULL;
+    }
+
+    Edge **edges = calloc(edges, sizeof(Edge *));
+    if (edges == NULL)
+    {
+        free(graph);
+        return NULL;
+    }
+    graph->edgeAmount = edgeAmount;
+    graph->edges = edges;
+}
+
+void connectNode(Graph *graph, size_t index, size_t nodeNumber)
+{
+    Edge *edge = graph->edges[index];
+    switch (edge->firstNodeNumber == 0)
+    {
+    case false:
+    {
+        edge->firstNodeNumber = nodeNumber;
+        break;
+    }
+
+    case true:
+    {
+        edge->secondNodeNumber = nodeNumber;
+        break;
+    }
+
+    default:
+        break;
+    }
```

Comment:
Лол.
```suggestion
    if (edge->firstNodeNumber == 0)
    {
        edge->secondNodeNumber = nodeNumber;
    } 
    else 
    {
        edge->firstNodeNumber = nodeNumber;
    }
```
А то default для булевого выражения выглядит очень мило, особенно с break, который ничего не делает.